src/Microsoft.Devices.HardwareDevCenterManager/Utility/BlobStorageHandler.cs

```diff
@@ -0,0 +1,84 @@
+﻿/*++
+    Copyright (c) Microsoft Corporation. All rights reserved.
+
+    Licensed under the MIT license.  See LICENSE file in the project root for full license information.  
+--*/
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Devices.HardwareDevCenterManager.Utility
+{
+    public class BlobStorageHandler
+    {
+        private readonly CloudBlockBlob blob;
+        private const int BlockSize = 256 * 1024;
+
+        /// <summary>
+        /// Handles upload and download of files for HWDC Azure Blob Storage URLs
+        /// </summary>
+        /// <param name="SASUrl">URL String to the blob</param>
+        public BlobStorageHandler(string SASUrl)
+        {
+            blob = new CloudBlockBlob(new Uri(SASUrl));
+        }
+
+        /// <summary>
+        /// Uploads specified file to HWDC Azure Storage
+        /// </summary>
+        /// <param name="filepath">Path to the file to upload to the Azure Blob URL</param>
+        /// <returns>True if the upload succeeded</returns>
+        public async Task<bool> Upload(string filepath)
+        {
+            fileSize = new System.IO.FileInfo(filepath).Length;
+            Console.Write("  0%");
```

Comment:
>Console.Write("  0%"); [](start = 12, length = 22)

can we send this to the `Debug` channel rather than standard output?

Reply:
is this the start of a progress bar on the console?

---
In reply to: [330293103](https://github.com/microsoft/devices-hardware-dev-center-manager/pull/1#discussion_r330293103) [](ancestors = 330293103)

Reply:
SDCM prints everything to the console as a CLI tool. I'm not sure what affect switching to Debug would have. I want to be able to keep SDCM functioning the same after this change.

Reply:
yes, i don't want any functional changes through all this refactoring either